components/suggestion/suggestion_box.jsx

```diff
@@ -617,10 +617,7 @@ export default class SuggestionBox extends React.PureComponent {
         const items = suggestions.items;
         let selection = this.state.selection;
         if (terms.length > 0) {
-            // if the current selection is no longer in the map, select the first term in the list
-            if (!this.state.selection || terms.indexOf(this.state.selection) === -1) {
-                selection = terms[0];
-            }
+            selection = terms[0];
```

Comment:
Something we're losing by changing this in this way is that the autocomplete would previously keep your cursor where it was if you select something and then continue typing. For example, if you type "t" and scroll to "Town Square" and then continue typing the rest of the channel name, it would keep Town Square selected even if it's not the first result that appears for "to".

We'll want to keep that, but we want to select the first result when the user hits backspace, so are you able to find a way to treat those differently when it comes to handling the selection?

Reply:
got it will work on that change

Reply:
@hmhealey don't you think the functionality should work while typing as well. Does not make sense to restrict this for only the backspace. Because when a new list comes obviously the 1st should be selected. Please share your thoughts on this. @nevyangelova also told me to just update the test case

Reply:
Yeah, I'm starting to agree with you there. Back when we first added the `SuggestionBox`, we thought it was important to keep the selection if the user continued typing, but I don't know if it makes sense any more. I've asked our UX team what their thoughts are on changing that in all cases.

Reply:
@hmhealey any updates on this. We can proceed with the new changes and get the code merged if you agree with me.

Reply:
It's been some time since this issue is around. We need to get it merged.